### ISSUE TITLE
[feature] lower-case option?

### ISSUE THREAD

**Issue by dvdplm** [jjpe/radix#2]
It would be nice to have the option to use lower-case letters when converting to string. In my case I use this to generate string keys for redis values and my aim is to be compatible with a previous javascript codebase. In Javascript `Number(123456789).toString(36)` outputs `21i3v9` while Radix gives `21I3V9`.

Thanks!

EDIT: fwiw, the `BigNum` crates `to_str_radix(36)` also uses lower-case.

**Comment by jjpe** [comment]
You can achieve this effect by converting the `str` e.g. `RadixNum::from(42 as u8).as_str().to_lowercase()`.  Does that work for your use case?

**Comment by dvdplm** [comment]
> .to_lowercase()

Absolutely, and that's what I do. It is not in a performance sensitive part of the code. It was more of a suggestion for other scenarios where the extra  allocation might be a problem.

**Comment by jjpe** [comment]
What you suggest can definitely be done, in principle, and if I had infinite resources I would definitely simply implement it.
However, my resources are limited (mostly time is relevant here), and I also have no concrete use case for the feature;  and given that `.to_lowercase()` is fast enough for your use case, I'd say that at least at this moment, neither do you.

If there is at least 1 active user that actually needs the extra efficiency, it can easily be a worthwhile feature to implement, but as it currently stands it looks like a bunch of effort (+ future maintenance costs) for not a lot of  concrete gain, at least from my perspective.
